File: examples-sdk/go/001-basic-ai-gw-demo/main.go
// 001-basic-ai-gw-demo — Go SDK equivalent
// Compile: vil compile --from go --input 001-basic-ai-gw-demo/main.go --release
package main

import vil "github.com/OceanOS-id/vil-go"

func main() {
	p := vil.NewPipeline("DecomposedPipeline", 3080)

	p.Sink(vil.SinkOpts{Name: "webhook_trigger", Port: 3080, Path: "/trigger"})
	p.Source(vil.SourceOpts{Name: "sse_inference", URL: "http://127.0.0.1:4545/v1/chat/completions", Format: "sse", JSONTap: "choices[0].delta.content", Dialect: "openai"})

	p.Route("sink.trigger_out", "source.trigger_in", "LoanWrite")
	p.Route("source.response_data_out", "sink.response_data_in", "LoanWrite")
	p.Route("source.response_ctrl_out", "sink.response_ctrl_in", "Copy")

	p.Compile()
}

File: examples-sdk/zig/001-basic-ai-gw-demo/main.zig
// 001-basic-ai-gw-demo — Zig SDK equivalent
// Compile: vil compile --from zig --input 001-basic-ai-gw-demo/main.zig --release

const vil = @import("vil");

pub fn main() void {
    var p = vil.Pipeline.init("DecomposedPipeline", 3080);
    p.sink("webhook_trigger", 3080, "/trigger");
    p.source("sse_inference", "http://127.0.0.1:4545/v1/chat/completions", "sse");
    p.route("sink.trigger_out", "source.trigger_in", "LoanWrite");
    p.route("source.response_data_out", "sink.response_data_in", "LoanWrite");
    p.route("source.response_ctrl_out", "sink.response_ctrl_in", "Copy");
    p.compile();
}
